meta/src/main/scala/scalan/meta/ScalanCodegen.scala

```diff
@@ -422,26 +360,125 @@ trait ScalanCodegen extends ScalanParsers { ctx: EntityManagement =>
         |""".stripAndTrim
       }
 
+      defs.mkString("\n\n")
+    }
+
+    def getTraitAbs = {
+      def entityProxy(entityName: String, typesDecl: String, typesUse: String) =
+        s"""
+        |  // single proxy for each type family
+        |  implicit def proxy$entityName${typesDecl}(p: Rep[$entityName${typesUse}]): $entityName$typesUse =
+        |    proxyOps[$entityName${typesUse}](p)
+        |""".stripAndTrim
+
+      def familyElem(entityName: String, tyArgsDecl: List[String], typesUse: String) =
+        s"""
+        |  abstract class ${entityName}Elem[${tyArgsDecl.opt(tyArgs => s"${tyArgsDecl.rep(t => t)}, ")}From, To <: $entityName$typesUse](iso: Iso[From, To]) extends ViewElem[From, To]()(iso) {
+        |    override def convert(x: Rep[Reifiable[_]]) = convert$entityName(x.asRep[$entityName$typesUse])
+        |    def convert$entityName(x : Rep[$entityName$typesUse]): Rep[To]
+        |  }
+        |""".stripAndTrim
+
+      //Define the list to put data into, could live with
+      //just a string but this way it is a bit more flexible
+      //In case of a large performance overhead switch to a string
+      val entitiesCode = mutable.MutableList[List[String]]()
+      for( entity <- module.entityOps ) {
```

Comment:
Yes, old code, I do not do these kind of things any more :) Must fix, will be good, I swear :D